### ISSUE TITLE
grunt failed on Windows environment

### ISSUE THREAD

**Issue by Kazuki-Nakanishi** [node-red/node-red#1178]
grunt command failed when I run it on Windows environment.
The stack trace is as follows.

```
   3) LocalFileSystem should return a newly saved library object:

      AssertionError: expected Array [] to equal Array [ Object { fn: 'file3.js', mno: 'pqr' } ] (at length, A has 0 and B has 1)
      + expected - actual

      -[]
      +[
      +  {
      +    "fn": "file3.js"
      +    "mno": "pqr"
      +  }
      +]
      
    at Assertion.fail (node_modules\should\lib\assertion.js:92:17)
    at Assertion.value (node_modules\should\lib\assertion.js:164:19)
    at test\red\runtime\storage\localfilesystem_spec.js:592:38
    at tryCatchReject (node_modules\when\lib\makePromise.js:845:30)
    at runContinuation1 (node_modules\when\lib\makePromise.js:804:4)
    at Fulfilled.when (node_modules\when\lib\makePromise.js:592:4)
    at Pending.run (node_modules\when\lib\makePromise.js:483:13)
    at Scheduler._drain (node_modules\when\lib\Scheduler.js:62:19)
    at Scheduler.drain (node_modules\when\lib\Scheduler.js:27:9)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by dceejay** [comment]
Ah - yes ... 
Some of the tests that are using the file system... like the storage api... are using linux style paths in the test... so the tests need to be fixed...

eg line 590 of localfilesystem_spec.js that is

     localfilesystem.saveLibraryEntry('object','B/D/file3.js',{mno:'pqr'},"// another non meta line\n\n Hi There").then(function() {

should be replaced by

    var ft = path.join("B","D","file3.js");
    localfilesystem.saveLibraryEntry('object',ft,{mno:'pqr'},"// another non meta line\n\n Hi There").then(function() {

That is just an example ... all pathed files should use `path.join` instead of Unix / syntax.

**Comment by Kazuki-Nakanishi** [comment]
Thank you for your fix.  But the problem still remains in my environment.
I looked into the problem and found that a file reading process started before finishing the file writing process.

One of the way to fix this problem is to use setTimeout although I'm not sure this approach is the best.
At least it works well in my environment.
The following fixed code is around the line 590 of localfilesystem_spec.js.
```
                localfilesystem.saveLibraryEntry('object',ft,{mno:'pqr'},"// another non meta line\n\n Hi There").then(function() {
                    setTimeout(function() {
                        localfilesystem.getLibraryEntry('object',path.join("B","D")).then(function(flows) {
                            flows.should.eql([ { mno: 'pqr', fn: 'file3.js' } ]);
                            localfilesystem.getLibraryEntry('object',ft).then(function(body) {
                                body.should.eql("// another non meta line\n\n Hi There");
                                done();
                            }).otherwise(function(err) {
                                done(err);
                            });
                        }).otherwise(function(err) {
                            done(err);
                        })}
                        , 100);
                }).otherwise(function(err) {
                    done(err);
                });
```

I remember you said that grunt sometimes fails due to the problem of timing.  This approach may also fix those problems.

**Comment by dceejay** [comment]
Hi - happy to accept a PR if you wish ?

**Comment by dceejay** [comment]
Just checking - you did get the latest version I assume - as I did take a couple of goes at it :-(...
If so then yes please to a PR.

**Comment by Kazuki-Nakanishi** [comment]
Yes, I did get the latest version.  I sent a PR.
Now I could pass all tests in simplemocha:core.

Although I ran into a new problem in simplemocha:nodes, this is an another problem.  I'll check it later.

**Comment by dceejay** [comment]
Ah yes... in the failing tests use 
    var os = require("os");
and then around the actual test check lines.
    if (os.type() === "Windows_NT") { ....
to test for the OS and do an appropriate test

**Comment by dceejay** [comment]
closed by 3658d0e03958ef6b8bcb5307646b172a8056c4dd
Thanks @Kazuki-Nakanishi
